### PR TITLE
NikitK/ Convert containers folder to ts 

### PR DESCRIPTION
**Pull request by NikitK-deriv**
Changes:

-   convert containers folder to ts

## Type of change

-   [ ] Bug fix
-   [ ] New feature
-   [x] Update feature
-   [ ] Refactor code
-   [ ] Translation to code
-   [ ] Translation to crowdin
-   [ ] Script configuration
-   [ ] Improve performance
-   [ ] Style only
-   [ ] Dependency update
-   [ ] Documentation update
-   [ ] Release

### CODE REVIEW

**Review by mohammad-rework — DISMISSED** [review]

File: src/components/containers/show.tsx [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    min_width?: string & number
+}
+
+export const Desktop = ({ children, max_width, ...props }: ResponseType) => (
+    <Media greaterThanOrEqual={max_width || 'tabletL'} {...props}>
+        {children}
+    </Media>
+)
+
+// We can't type it properly due fresnel
```
Comment (aswathy-deriv) [inline src/components/containers/show.tsx:17]: is this comment needed?

File: src/components/containers/show.tsx [hunk 2]
```diff
@@ -0,0 +1,45 @@
+import React, { useContext, ReactNode } from 'react'
+import { Media } from 'themes/media'
+import { DerivStore } from 'store'
+
+type ResponseType = {
+    children?: ReactNode
+    max_width?: string & number
```
Comment (aswathy-deriv) [inline src/components/containers/show.tsx:7]: we can use max_width:string | number right

File: src/components/containers/show.tsx [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    min_width?: string & number
```
Comment (aswathy-deriv) [inline src/components/containers/show.tsx:8]: here also.instead of and operator use or

File: src/components/containers/show.tsx [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+    const { is_eu_country } = useContext(DerivStore)
+
+    if (is_eu_country) return <>{children}</>
+    else return null
+}
+
+export const NonEU = ({ children }: DomainType) => {
+    const { is_eu_country } = useContext(DerivStore)
+
+    if (is_eu_country === false) return <>{children}</>
```
Comment (maryia-binary) [inline src/components/containers/show.tsx:36]: small suggestion:
```suggestion
    if (!is_eu_country) return <>{children}</>
```
is it gonna work if you `return children` without wrapping it in a fragment <></> ?  😊

File: src/components/containers/show.tsx (hunk 2 above)
Comment (NikitK-deriv) [reply src/components/containers/show.tsx:7]: Unfortunately no.
We use dependency fresnel which should include `greaterThanOrEqual` which include string and number type in node_modules. I'm not yet realized how to change it
Screenshot 2022-07-28 at 15 00 41

File: src/components/containers/show.tsx (hunk 1 above)
Comment (NikitK-deriv) [reply src/components/containers/show.tsx:16]: Seems we should explain why we decide to use any here

File: src/components/containers/show.tsx (hunk 4 above)
Comment (NikitK-deriv) [reply src/components/containers/show.tsx]: Resolved as NonEU = Eu
Because components are the same 💩

**Review by mohammad-rework — APPROVED** [review]

**Pull request merged**